### DISCUSSION TITLE
exp, exp2 functions are not working as expected

### DISCUSSION THREAD

**Discussion by felix0888** [PaulRBerg/prb-math#66] [Q&A] [answered]
Trying the exp, exp2 functions, I noticed that those functions are not working the way I expected.

Actually, I was expecting `x.exp` to return `e^x` with some zeros, but not. I send you some test data and the result I'm getting.
```solidity
uint x = 1; // test input
return x.exp();
```

```
x                 x.exp()
1                 1000000000000000000
2                 1000000000000000002
3                 1000000000000000002
10                1000000000000000009
100               1000000000000000099
...
```

Similar result when trying `exp2` function.
Any misunderstanding on my side?

**Comment by PaulRBerg** [comment, marked as answer]
> Any misunderstanding on my side?

Yup. This is a **fixed-point** math library, which means that numbers are assumed to have 18 decimals. Always.

In your case, when  you set `x` to `1`, in normal numbers that's actually `0.000000000000000001`. The result you got is correct: taking `e` to the power of `0.000000000000000001` gives you ~1.

Please refer to the README, this is explained thoroughly there.

**Reply by felix0888** [reply]
Absolutely, I got what you mean.
Thank you for your quick help.

**Reply by PaulRBerg** [reply]
Happy to help.
